rclcpp/src/rclcpp/parameter_client.cpp

```diff
@@ -273,6 +273,58 @@ AsyncParametersClient::set_parameters_atomically(
   return future_result;
 }
 
+std::shared_future<std::vector<rcl_interfaces::msg::SetParametersResult>>
+AsyncParametersClient::delete_parameters(
+  const std::vector<std::string> & parameters_names)
+{
+  std::vector<rclcpp::Parameter> parameters;
+  for (const std::string & name : parameters_names) {
+    parameters.push_back(rclcpp::Parameter(name));
+  }
+  auto future_result = set_parameters(parameters);
+
+  return future_result;
+}
+
+std::shared_future<std::vector<rcl_interfaces::msg::SetParametersResult>>
+AsyncParametersClient::load_parameters(
+  const std::string & yaml_filename)
+{
+  // parse yaml file
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rcl_params_t * rcl_parameters = rcl_yaml_node_struct_init(allocator);
+  const char * path = yaml_filename.c_str();
+  if (!rcl_parse_yaml_file(path, rcl_parameters)) {
+    throw std::runtime_error("Unable to get get parameters from yaml file");
+  }
+
+  // create list of parameters to set
+  rclcpp::ParameterMap parameter_map = rclcpp::parameter_map_from(rcl_parameters);
+  std::vector<rclcpp::Parameter> parameters;
+  std::string fully_qualified_name =
+    node_topics_interface_->get_node_base_interface()->get_fully_qualified_name();
+
+  // only add remote_node parameters
+  for (auto & params : parameter_map) {
+    std::string node_name = params.first;
+    if (node_name == fully_qualified_name ||
+      node_name == "/**" ||
+      (node_name.substr(node_name.find("/*/") + 3) == remote_node_name_))
+    {
+      for (auto & param : params.second) {
+        parameters.push_back(param);
+      }
+    }
+  }
+
+  if (parameters.size() == 0) {
+    throw std::runtime_error("No valid parameters from yaml file");
```

Comment:
Should this be an exception or maybe should it silently pass if there are no parameter matching the target node? Either way this is the kind of thing that needs to be documented in the header.

Reply:
To pass silently, one has to fix the crash that happens if set_parameters is called with an empty vector.

Reply:
Maybe I don't understand, but could you just not call `set_parameters` at all in this case? I guess then you wouldn't know what to return... In that case I guess throwing an exception here is ok.